### PR TITLE
better trending APIs + homepage

### LINKED ISSUE #7378: Tweak trending API: Sort by date
The Trending carousel biases towards the same books, creating a re-enforcing algorithm. Use the date to help randomize results (i.e. sort most logged books by date instead of count)

### Describe the problem that you'd like solved

https://github.com/internetarchive/openlibrary/blob/fd990d519672ab74c3ba4a45916b504622d53d12/openlibrary/views/loanstats.py#L49-L65
and
https://github.com/internetarchive/openlibrary/blob/fd990d519672ab74c3ba4a45916b504622d53d12/openlibrary/core/bookshelves.py#L88-L107

Add a parameter to `most_logged_books` to sort most logged books secondarily by date (which sorts results by date rather than most logged). See: https://stackoverflow.com/questions/2051162/sql-multiple-column-ordering

### PR DESCRIPTION
**Pull request by mekarpeles**
Closes #7378

### CODE REVIEW

File: openlibrary/core/bookshelves.py [hunk 1]
```diff
@@ -106,8 +109,21 @@ def most_logged_books(
...
+        orders = ['cnt desc']
+        if distinct:
+            orders = ['cnt desc'] + orders
+            distinct = 'distinct on (work_id)'
+        order_by = f'order by {", ".join(orders)}'
+        if not sort_by_count:
+            order_by = ''
+        select = (
+            f'select {distinct or ""} work_id, count(*) as cnt from bookshelves_books'
+        )
```
Comment (cclauss) [inline openlibrary/core/bookshelves.py:125]: Avoid converting a bool to a str.
```suggestion
            distinct_str = 'distinct on (work_id)'
        else:
            distinct_str = ''
        order_by = f'order by {", ".join(orders)}'
        if not sort_by_count:
            order_by = ''
        select = (
            f'select {distinct_str} work_id, count(*) as cnt from bookshelves_books'
        )
```

**Comment by mekarpeles** [comment]
This is also a potential approach of clamping the work count (though it seems to be resulting in more dupes)

```
select work_id, CASE
                WHEN COUNT(work_id) > 7 THEN 7
                ELSE COUNT(work_id)
            END as cnt
            from bookshelves_books
            WHERE bookshelf_id IS NOT NULL  AND created >= $since group by work_id HAVING COUNT(*) > 3 order by cnt desc
            limit $limit offset $offset
```

File: openlibrary/plugins/openlibrary/api.py [hunk 2]
```diff
@@ -71,14 +71,18 @@ def GET(self, period="/daily"):
         from openlibrary.views.loanstats import SINCE_DAYS
 
         period = period[1:]  # remove slash
-        i = web.input(page=1, limit=100, days=0, hours=0)
+        i = web.input(
+            page=1, limit=100, days=0, hours=0, sort_by_count=False, minimum=''
```
Comment (cdrini) [inline openlibrary/plugins/openlibrary/api.py:75]:
```suggestion
            page=1, limit=100, days=0, hours=0, sort_by_count=False, minimum=0
```

**Review by cdrini — APPROVED** [review]
Lgtm, tested on mek's box. Ran without a hitch. The algorithm is slightly too much deprioritizing trending books now; the 10 clamping worked nicely, but caused duplicates. But this looks better now than before and provides more variety!

**Pull request merged**